source/common/filter/tcp_proxy.h

```diff
@@ -104,6 +105,13 @@ class TcpProxyConfig {
   const Optional<std::chrono::milliseconds>& idleTimeout() { return shared_config_->idleTimeout(); }
   TcpProxyUpstreamDrainManager& drainManager();
   SharedConfigSharedPtr sharedConfig() { return shared_config_; }
+  const Router::MetadataMatchCriteria* metadataMatchCriteria() {
+    if (cluster_metadata_match_criteria_) {
```

Comment:
No need to check this pointer.  Just "return cluster_metadata_match_criteria_.get();" which will return nullptr if the unique_ptr is null